### PR TITLE
V1 Window/Ledger based on RocksDb

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Current window implementation doesn't scale for large gaps/window sizes. The ledger also doesn't account for forks.

#### Summary of Changes
Implement basic ledger module based on RocksDb, doesn't yet account for forks. This v1 will first be used to replace the existing version of window before proceeding to v2.

Fixes #
Nothing yet.

### CODE REVIEW

File: src/db_ledger.rs [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+pub struct DataCf {
+    handle: ColumnFamily,
+}
+
+impl DataCf {
+    pub fn new(handle: ColumnFamily) -> Self {
+        DataCf { handle }
+    }
+
+    pub fn key(slot_height: u64, index: u64) -> Vec<u8> {
```
Comment (rob-solana) [inline src/db_ledger.rs]: slot_height? is that leader period?

Comment (carllin) [reply src/db_ledger.rs]: yup!

**Comment by rob-solana** [comment]
cc #1717

File: src/db_ledger.rs [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+        // Destroying database without closing it first is undefined behavior
+        drop(ledger);
+        let _ignored = DB::destroy(&Options::default(), &ledger_path);
+    }
+
+    #[test]
+    fn test_process_new_blobs_basic() {
+        let shared_blobs = make_tiny_test_entries(2).to_blobs();
+        let blob_locks: Vec<_> = shared_blobs.iter().map(|b| b.read().unwrap()).collect();
+        let blobs: Vec<&Blob> = blob_locks.iter().map(|b| &**b).collect();
```
[the remark this answers is not in the document]
Comment (carllin) [reply src/db_ledger.rs]: blob_locks is a vector of RwLockReadGuard<Blob>, so blob_locks.iter() returns &RwLockReadGuard<Blob>. You dereference once to get RwLockReadGuard<Blob>, then dereference again to get Blob, which we then need a reference to.

[the remark this answers is not in the document]
Comment (rob-solana) [reply src/db_ledger.rs]: ah...  equivalent to borrow() on the rwlockreadguard?

[the remark this answers is not in the document]
Comment (carllin) [reply src/db_ledger.rs]: haha yeah, didn't know that existed.

File: ci/docker-rust/Dockerfile [hunk 3]
```diff
@@ -12,6 +12,7 @@ RUN set -x && \
     apt update && \
     apt install -y \
       buildkite-agent \
+      clang-7 \
```
Comment (garious) [inline ci/docker-rust/Dockerfile]: Why exactly?

**Pull request merged**